### DISCUSSION TITLE
How to store the messages from data channel?

### DISCUSSION THREAD

**Discussion by mekya** [ant-media/Ant-Media-Server#5300] [FAQs] [answered]
How can you store the messages from the data channel in Mongodb?
As far as I can see, you'd need to use the data_received callback in the webrtc adaptor to trigger a process to send the message to a database. Is this the only way?

**Comment by tremzo** [comment, marked as answer]
There is an application setting `dataChannelWebHookURL` which will relay all data channel messages to the webhook URL that is defined.

The messages will only start getting sent once there is more than one connection on the live stream, for example, 2+ data channel only connections on a live stream or 1 broadcaster and 1+ viewer on a live stream. Otherwise, nothing is sent. 

Images that are sent in the data channel are relayed as binary parts, so a single image may send multiple binary messages as you can see in the following screenshot:

<img width="1718" alt="Screenshot 2023-06-27 at 12 17 07" src="https://github.com/ant-media/Ant-Media-Server/assets/121940452/8ca0728b-796d-42a5-ba2e-66e1f4e5c61c">



Here is an example POST request sending a text message through the data channel:


<img width="1655" alt="Screenshot 2023-06-27 at 12 16 55" src="https://github.com/ant-media/Ant-Media-Server/assets/121940452/f3cc0026-d08a-40db-9a23-c19b9d439657">


Here is the setting that has been configured in the application settings file `red5-web.properties` using the new advanced section in the web panel now available in version 2.6.2

<img width="1712" alt="Screenshot 2023-06-27 at 12 23 44" src="https://github.com/ant-media/Ant-Media-Server/assets/121940452/87b02401-1a27-46f3-9467-a4a1e28e8365">

**Reply by mekya** [reply]
Thank you @timantmedia
